neuros/process_data.py

```diff
@@ -1,114 +1,63 @@
+    Get frequency range for a specific brain wave band.
 
     Args:
-        data: 1D numpy array of samples
-        sampling_rate: Sampling rate in Hz
+        band: Brain wave band.
 
     Returns:
-        Dictionary of band names to power values
+        Tuple of (low_freq, high_freq) for the specified band.
     """
-    bands = {
-        'delta': (1, 4),
-        'theta': (4, 8),
-        'alpha': (8, 13),
-        'beta': (13, 30),
-        'gamma': (30, 50),
-        'total': (1, 50)
-    }
-
-    return {
-        name: extract_band_power(data, sampling_rate, low, high)
-        for name, (low, high) in bands.items()
-    }
+    if band == Band.DELTA:
+        return 0.5, 4.0
+    if band == Band.THETA:
+        return 4.0, 8.0
+    if band == Band.ALPHA:
+        return 8.0, 13.0
+    if band == Band.BETA:
+        return 13.0, 30.0
+    if band == Band.GAMMA:
+        return 30.0, 100.0
+    if band == Band.ALL:
+        return 0.5, 100.0
+    raise ValueError(f"Invalid band: {band}")
 
 
-def compute_band_ratios(powers: Dict[str, float]) -> Dict[str, float]:
+def compute_power(channel_data: np.ndarray, sampling_rate: int, band: Band) -> float:
     """
-    Compute standard ratios between frequency bands.
+    Extract band power from EEG data.
 
     Args:
-        powers: Dictionary of band powers from extract_all_bands()
+        channel_data: 1D numpy array of samples.
+        sampling_rate: Sampling rate in Hz.
+        band: Brain wave band.
 
     Returns:
-        Dictionary of ratio names to values
+        Band power as a float.
     """
-    ratios = {
-        'alpha_theta': ('alpha', 'theta'),
-        'alpha_beta': ('alpha', 'beta'),
-        'theta_beta': ('theta', 'beta'),
-        'alpha_total': ('alpha', 'total')
-    }
-
-    return {
-        name: powers[num] / (powers[den] + 1e-10)
-        for name, (num, den) in ratios.items()
-    }
-
-
-class PowerMetrics(NamedTuple):
-    """Collection of power measurements for a channel"""
-    absolute_alpha: float
-    alpha_ratio: float  # alpha/total
-    alpha_beta_ratio: float
-
-
-def process_channel(data: np.ndarray, sampling_rate: int) -> PowerMetrics:
-    """
-    Process a single channel of EEG data to extract power metrics.
-
-    Args:
-        data: 1D numpy array of samples
-        sampling_rate: Sampling rate in Hz
-
-    Returns:
-        PowerMetrics containing absolute and relative measurements
-    """
-    # Get all band powers
-    powers = extract_all_bands(data, sampling_rate)
-    ratios = compute_band_ratios(powers)
-
-    return PowerMetrics(
-        absolute_alpha=powers['alpha'],
-        alpha_ratio=ratios['alpha_total'],
-        alpha_beta_ratio=ratios['alpha_beta']
+    low_freq, high_freq = get_band_range(band)
+    filtered = channel_data.copy()
+    DataFilter.detrend(filtered, DetrendOperations.CONSTANT.value)
+    DataFilter.perform_bandpass(
+        filtered,
+        sampling_rate,
+        low_freq,
```

Comment:
If the fields of Band are named, then then this line now reads better as `band.low_freq`